src/Util.h

```diff
@@ -185,6 +186,29 @@ std::string replace_all(const std::string &str, const std::string &find, const s
 /** Split the source string using 'delim' as the divider. */
 std::vector<std::string> split_string(const std::string &source, const std::string &delim);
 
+/** Join the source vector using 'delim' as the divider. */
+template<typename T>
```

Comment:
This is only going to work for strings now, so there's no point in templating it

Reply:
could work for `char *` and `string_view` too

Reply:
Fair enough. It requires a .size() so not char *, but I guess it would work with string_view